flake.nix

```diff
@@ -7,13 +7,22 @@
       url = "github:nix-community/poetry2nix";
       inputs.nixpkgs.follows = "nixpkgs";
     };
+    legacy-nixpkgs.url = "github:nixos/nixpkgs/b511bf59bd6754ad49c637f390dbb6bc1acea000";
   };
-  outputs = { self, nixpkgs, poetry2nix }@inputs:
+  outputs = { self, nixpkgs, poetry2nix, legacy-nixpkgs }@inputs:
     let
       supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
       forSystems = systems: f:
         nixpkgs.lib.genAttrs systems
-        (system: f system (import nixpkgs { inherit system; overlays = [ poetry2nix.overlay self.overlays.default ]; }));
+        (system: f system (import nixpkgs { 
+          inherit system; 
+          overlays = [ poetry2nix.overlay self.overlays.default ]; 
+          # This is needed for the dependency "cryptography" of cashu 
+          # which is built with python36 which is using openssl_1_1 injected by peotry2nix's defaultOverrides
+          config.permittedInsecurePackages = [
```

Comment:
Let's not deal with this here, but let's fix cashu upstream and then use the fixed version of the upstream.